src/journal/urls.py

```diff
@@ -7,15 +7,26 @@
 
 from journal import views
 
+from identifiers.models import IDENTIFIER_TYPES
+
+idents = "|".join(IDENTIFIER_TYPES)
+
 urlpatterns = [
-    # Probably needs some multi-journal logic here
+    # Figures and download patterns
+    url(r'^article/(?P<identifier_type>{0})/(?P<identifier>.+)/print/$'
+        ''.format("|".join(IDENTIFIER_TYPES)),
+        views.print_article,
+        name='article_print_article'),
+    url(r'^article/(?P<identifier_type>doi)/(?P<identifier>10.\d{4,9}/[-._;()/:A-Za-z0-9]+)/print/$',
+        views.print_article,
+        name='article_print_article'),
     url(r'^article/(?P<article_id>\d+)/galley/(?P<galley_id>\d+)/figure/(?P<file_name>.*)/$',
         views.article_figure,
-        name='article_figure'),
-    url(r'^article/(?P<identifier_type>.+?)/(?P<identifier>.+)/file/(?P<file_id>\d+)/replace$',
+        name='article_galley_figure'),
+    url(r'^article/(?P<identifier_type>id)/(?P<identifier>.+)/file/(?P<file_id>\d+)/replace$',
         views.replace_article_file,
         name='article_file_replace'),
-    url(r'^article/(?P<identifier_type>.+?)/(?P<identifier>.+)/file/(?P<file_id>\d+|None)/$',
+    url(r'^article/(?P<identifier_type>id)/(?P<identifier>.+)/file/(?P<file_id>\d+|None)/$',
```

Comment:
Why do we need to support the file id `None` here but not in `article_file_replace` above?

Reply:
I believe I found why :grimacing: 
https://github.com/BirkbeckCTP/janeway/blob/39acd7a83eefa4b23076b3f36deb0600aea3c88c/src/journal/views.py#L426

Reply:
mauro.sanchez Today at 11:07
if something works, don't touch it, or so they say

Reply:
haha I believe it was "if ain't broke don't fix-it"